File: Backend/Models/Business.cs
namespace SmartOfferBookingSystem.Models;

public sealed class Business
{
    public Guid Id { get; set; }
    public Guid OwnerId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? PhoneNumber { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public User Owner { get; set; } = null!;
    public ICollection<Offer> Offers { get; set; } = [];
}

File: Frontend/src/types/business.ts
export interface BusinessSummary {
  id: string
  name: string
  slug: string
  description: string | null
  phoneNumber: string | null
  createdAt: string
  updatedAt: string
}

export interface UpsertBusinessPayload {
  name: string
  slug: string
  description?: string | null
  phoneNumber?: string | null
}
